src/main/webapp/js/reducers/field-config-reducer.js

```diff
@@ -9,13 +9,20 @@ import {
   CHANGE_FIELD_VISIBILITY,
   CHANGE_FIELD_ORDER,
   SAVE_FIELD_CONFIG_ORDER,
+  FETCH_ALL_FIELD_CONFIGS,
+  START_FETCH_CONFIGS,
+  FINISH_FETCH_CONFIGS,
 } from '../actions/types';
+import roleFieldConfig from '../utils/field-configs';
 
 const initialState = {
   vaccinee: {},
   keyCommunityPerson: {},
   site: {},
   visit: {},
+  role: roleFieldConfig,
+  fetching: false,
+  fetched: false,
```

Comment:
do you need both fetching and fetched? I think fetched should be enough, it's only set to true after the configs finished fetching